environments/workspaces-as-code/templates.md

```diff
@@ -0,0 +1,206 @@
+**workspace templates**.
+
+Workspace templates are written as YAML and have a `.yaml` or `.yml` extension.
+Coder looks for your workspace template at the following path:
+
+```text
+<repository-root>/.coder/coder.yaml
+```
+
+![Template Location](../../assets/wac-location.png)
+
+## Workspace template sample
+
+The following is a sample workspace template that makes use of all available
+fields. Depending on your use case, you may not need all of the options
+available.
+
+For detailed information on the fields available, see the
+[subsequent sections](#workspace-template-fields) of this article
+
+```yaml
+version: 0.1
+workspace:
+  type: kubernetes
+  spec:
+    image: index.docker.io/ubuntu:18.04
+    container-based-vm: true
+    cpu: 4
+    memory: 16
+    disk: 128
+    gpuCount: 1
+    labels:
+      com.coder.custom.hello: "hello"
+      com.coder.custom.world: "world"
+    tolerations:
+      - key: my-key
+        operator: Equal
+        value: my-value
+        effect: NoExecute
+        tolerationSeconds: 3600
+  configure:
+    - name: "install curl"
+      run: |
+        apt update
+        apt install -y curl
+    - name: "install Go binary"
+      run: "go install"
+      directory: /home/coder/go/src/github.com/my-project
+      shell: "bash"
+      env:
+        GOPATH: /home/coder/go
+```
+
+## Workspace template fields
+
+### version
+
+The version number of the config file being used. The current version is `0.1`.
+
+### workspace
+
+**Required**. The section containing all configuration information related to
+the environment.
+
+#### workspace.type
+
+**Required**. Determines the type of workspace to be created. Currently, the
+only accepted value is `kubernetes`.
+
+#### workspace.spec
+
+**Required**. This section contains configuration information specific to the
+`workspace.type`.
+
+#### workspace.spec.image
+
+**Required**. The image to use for the environment. The image should include the
+registry and (optionally) the tag, i.e. `docker.io/ubuntu:18.04`. If you omit
+the tag, Coder uses the default value of `latest`.
+
+You must have [imported the image](../../images/importing.md) into Coder,
+otherwise the environment will fail to build.
+
+#### workspace.spec.labels
+
+The
+[Kubernetes labels](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/)
+to be added to the environment pod.
+
+```yaml
+workspace:
+  labels:
+    com.coder.custom.hello: hello
+    com.coder.custom.world: world
+```
+
+#### workspace.spec.tolerations
+
+This section defines the
+[Kubernetes tolerations](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/)
+to be added to the pod hosting the environment.
+
+```yaml
+workspace:
+  tolerations:
+    - key: my-key
+      operator: equals
+      value: my-value
+      effect: NoExecute
+      tolerationSeconds: 3600
+```
+
+#### workspace.spec.gpucount
+
+The number of GPUs to inject into the environment.
```

Comment:
I think inject is the wrong term here but may be wrong. I'd say "allocate" if it were my sentence.